### PR TITLE
test: move instances tests to "tests-integration"

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

The testing codes in crate "frontend" create a complete distributed instance. Moving them to crate "tests-integration" so we can reuse them in other testing cases. For example, testing region failover.

Besides, it greatly reduces the codes complexity of frontend, making it easy to read. And what's more, I found that the original testing cases in "tests-integration" only test standalone instance. Maybe we could add other cases in it to test distributed instance, too.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/frontend/src/datanode.rs [hunk 1]
```diff
@@ -62,8 +62,8 @@ impl DatanodeClients {
             .await
     }
 
-    #[cfg(test)]
-    pub(crate) async fn insert_client(&self, datanode: Peer, client: Client) {
+    // Should be used only in tests.
```
Comment (evenyag) [inline src/frontend/src/datanode.rs:65]: What about adding a test feature?

File: src/frontend/src/instance.rs [hunk 2]
```diff
@@ -236,8 +236,7 @@ impl Instance {
         Ok(())
     }
 
-    #[cfg(test)]
-    pub(crate) async fn new_distributed(
+    pub async fn new_distributed(
```
[the remark this answers is not in the document]
Comment (MichaelScofield) [reply src/frontend/src/instance.rs:239]: I'll remove it completely in the following PR.

**Comment by MichaelScofield** [comment]
@evenyag PTAL

**Review by sunng87 — APPROVED** [review]

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**